code/datums/controllers/action_controls.dm

```diff
@@ -346,6 +346,76 @@ var/datum/action_controller/actions
 			del(icon_image)
 		..()
 
+/datum/action/bar/icon/callback //calls a specified proc if it finishes without interruptions, moderately customiseable
+	id = null //set to a string version of the callback proc path
+	interrupt_flags = INTERRUPT_MOVE | INTERRUPT_ACT | INTERRUPT_STUNNED | INTERRUPT_ACTION //unsure if these will need to be overriden
+	var/proc_path = null //set to the proc path of the proc to be "called back"
+	var/target = null //what the target of the action is, if any
+	var/end_message = ""
+	var/maximum_range = 1 //what is the maximum range target and owner can be apart? need to modify directly before starting the action.
+	var/list/proc_args = null //a list of args for the callback proc, if needed.
```

Comment:
can you switch these to use the doc comment format

Reply:
done!